### PR TITLE
Ingest Campaign, Case and related sobjects

### PR DESCRIPTION
**Pull request by navarone-feekery**
## Closes https://github.com/elastic/enterprise-search-team/issues/5441

Ingest the final sobjects and all related fields:

- Campaign
- Case
  - CaseComments
  - EmailMessages
  - CaseFeeds
    - FeedComments

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: connectors/sources/salesforce.py [hunk 1]
```diff
@@ -332,6 +415,16 @@ async def _yield_non_bulk_query_pages(self, soql_query):
...
 
+    async def _get_non_bulk_query(self, soql_query):
+        """For quick queries, ignores pagination"""
+        url = f"{self.base_url}{QUERY_ENDPOINT}"
+        params = {"q": soql_query}
+        response = await self._get_json(
+            url,
+            params=params,
+        )
+        return response.get("records")
```
Comment (artem-shelkovnikov) [inline connectors/sources/salesforce.py:426]:
```suggestion
    async def _execute_non_paginated_query(self, soql_query):
        """For quick queries, ignores pagination"""
        url = f"{self.base_url}{QUERY_ENDPOINT}"
        params = {"q": soql_query}
        response = await self._get_json(
            url,
            params=params,
        )
        return response.get("records")
```

File: connectors/sources/salesforce.py [hunk 2]
```diff
@@ -773,6 +1076,104 @@ def _format_address(self, address):
         ]
         return ", ".join([a for a in address_fields if a])
 
+    def _format_time_for_sort(self, time_body_pair):
+        if time_body_pair[0] is None:
+            return ""
+
+        return time_body_pair[0]
```
Comment (artem-shelkovnikov) [inline connectors/sources/salesforce.py:1083]: You can inline this function into the method that uses it, if it's the only place, it simplifies things a bit IMO

File: connectors/sources/salesforce.py [hunk 3]
```diff
@@ -773,6 +1076,104 @@ def _format_address(self, address):
...
+                time_body_pairs.append(
+                    [comment.get("CreatedDate"), comment.get("CommentBody")]
+                )
+
+        time_body_pairs = sorted(time_body_pairs, key=self._format_time_for_sort)
+        return "\n\n".join(
+            filter(
+                None, [str(x[-1]).strip() for x in time_body_pairs if x[-1] is not None]
+            )
+        )
```
Comment (artem-shelkovnikov) [inline connectors/sources/salesforce.py:1120]: What's going on here? Might be worth to make a function or add a comment explaining this

**Review by artem-shelkovnikov — COMMENTED** [review]
LGTM overall, left a couple comments/questions

File: connectors/sources/salesforce.py (hunk 3 above)
Comment (navarone-feekery) [reply connectors/sources/salesforce.py:1120]: Yeah this was much prettier in the original ruby code. I'll make it easier to understand.

Comment (navarone-feekery) [reply connectors/sources/salesforce.py]: @artem-shelkovnikov I broke it up and added comments

File: connectors/sources/salesforce.py (hunk 2 above)
Comment (navarone-feekery) [reply connectors/sources/salesforce.py]: Made it an in-line lambda 👍🏻

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**